src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -159,4 +163,39 @@ private Future<Void> validateLocationRestrictions(CompositePoLine poLine, List<F
     return Future.failedFuture(new HttpException(422, ErrorCodes.FUND_LOCATION_RESTRICTION_VIOLATION, parameters));
   }
 
+  private Set<String> extractRestrictedLocations(CompositePoLine poLine, Set<Fund> fundsWithRestrictedLocations, RequestContext requestContext) {
+    Set<String> validLocations = poLine.getLocations().stream().map(Location::getLocationId).filter(Objects::nonNull).collect(Collectors.toSet());
+    List<String> holdingIds = poLine.getLocations().stream().map(Location::getHoldingId).filter(Objects::nonNull).collect(Collectors.toList());
+    List<String> permanentLocationIdFromHoldings = getPermanentLocationIdFromHoldings(holdingIds, requestContext);
+    validLocations.addAll(permanentLocationIdFromHoldings);
+
+    // Checking fund location against validLocations in POL to identify restricted locations
+    // if there is one, will be stored to put in error as parameter
+    return fundsWithRestrictedLocations.stream()
+      .flatMap(fund -> fund.getLocationIds().stream())
+      .filter(locationId -> !validLocations.contains(locationId))
+      .collect(Collectors.toSet());
+  }
+
+  private List<String> getPermanentLocationIdFromHoldings(List<String> holdingIds, RequestContext requestContext) {
```

Comment:
need to rename to getPermanentLocationIdsFromHoldings

Reply:
thanks, renamed